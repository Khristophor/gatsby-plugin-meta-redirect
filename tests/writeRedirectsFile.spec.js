const { exists, ensureDir, remove, readFile } = require('fs-extra');
const writeRedirects = require('../src/writeRedirectsFile');

describe('writeRedirectsFile', () => {
  const tempFolderPath = './tmp';

  const assertRedirectFile = async expectedPath => {
    expect(await exists(expectedPath)).toBe(true);
    expect(await readFile(expectedPath, 'utf-8')).toMatchSnapshot(true);
  };

  beforeEach(async () => {
    await remove(tempFolderPath);
    await ensureDir(tempFolderPath);
  });

  // cleanup
  afterAll(async () => {
    await remove(tempFolderPath);
  });

  it('writes redirects', async () => {
    await writeRedirects(
      [
        {
          fromPath: '/',
          toPath: '/hello'
        }
      ],
      tempFolderPath
    );

    await assertRedirectFile(`${tempFolderPath}/index.html`);
  });

  it('writes redirects', async () => {
    await writeRedirects(
      [
        {
          fromPath: '/hello',
          toPath: '/'
        }
      ],
      tempFolderPath
    );

    await assertRedirectFile(`${tempFolderPath}/hello/index.html`);
  });

  it('writes deep path redirects', async () => {
    await writeRedirects(
      [
        {
          fromPath: '/a/b/c/d',
          toPath: '/x/y/z'
        }
      ],
      tempFolderPath
    );

    await assertRedirectFile(`${tempFolderPath}/a/b/c/d/index.html`);
  });

  it('handles external redirects', async () => {
    await writeRedirects(
      [
        {
          fromPath: '/a/b',
          toPath: 'http://example.com/'
        }
      ],
      tempFolderPath
    );

    await assertRedirectFile(`${tempFolderPath}/a/b/index.html`);
  });
});
