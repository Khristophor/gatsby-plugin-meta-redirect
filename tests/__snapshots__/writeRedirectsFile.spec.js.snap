// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeRedirectsFile handles external redirects: true 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='http://example.com/'\\" />"`;

exports[`writeRedirectsFile writes deep path redirects: true 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/x/y/z/'\\" />"`;

exports[`writeRedirectsFile writes redirects: true 1`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='/hello/'\\" />"`;

exports[`writeRedirectsFile writes redirects: true 2`] = `"<meta http-equiv=\\"refresh\\" content=\\"0; URL='//'\\" />"`;
